### PR TITLE
refac: Use one method for creating different integration events

### PR DESCRIPTION
**Pull request by dstenroejl**
# Event factory
Current design of `CalculationResultIntegrationEventFactory` will require that we add a new method whenever we add a new integration event or calculation result.

By using OO principles and adding and implementing an interface (`IEventMessage`) to the protobuf generated message class we can implement a generic method for creating an integration event based on the message.

# Event Provider
The current design of `IntegrationEventProvider` would also require changes whenever we change either an EnergyResult, WholesaleResult or any of their related events.

Splitting these two path's into separate implementation (`EnergyResultEventProvider` and `WholesaleResultEventProvider`) reduce the likelihood of having to change `IntegrationEventProvider` as often.

### CODE REVIEW

File: source/dotnet/wholesale-api/WebApi/Configuration/Modules/EventsRegistration.cs [hunk 1]
```diff
@@ -40,7 +41,8 @@ public static class EventsRegistration
         serviceCollection.AddScoped<ICompletedBatchFactory, CompletedBatchFactory>();
         serviceCollection.AddScoped<IRegisterCompletedBatchesHandler, RegisterCompletedBatchesHandler>();
 
-        serviceCollection.AddScoped<ICalculationResultIntegrationEventFactory, CalculationResultIntegrationEventFactory>();
+        serviceCollection.AddScoped<IEnergyResultEventProvider, EnergyResultEventProvider>();
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/WebApi/Configuration/Modules/EventsRegistration.cs:44]: @JesperGomes should we move these registrations into the "AddApplication" method or similar?

File: source/dotnet/wholesale-api/Events/Events.Application/Communication/EventProviderState.cs
Comment (dstenroejl) [inline source/dotnet/wholesale-api/Events/Events.Application/Communication/EventProviderState.cs:1]: Maybe we want to delete this class and just revert back to counting within `IntegrationEventProvider`

Comment (dstenroejl) [reply source/dotnet/wholesale-api/Events/Events.Application/Communication/EventProviderState.cs:1]: Fixed

File: source/dotnet/wholesale-api/WebApi/Configuration/Modules/EventsRegistration.cs (hunk 1 above)
Comment (JesperGomes) [reply source/dotnet/wholesale-api/WebApi/Configuration/Modules/EventsRegistration.cs:44]: Yes, I think we should

Comment (dstenroejl) [reply source/dotnet/wholesale-api/WebApi/Configuration/Modules/EventsRegistration.cs]: Fixed

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**